Add `Key::CTRL_OR_CMD` and use it to fix shortcut for tilemap painting tools on macOS

ajreckof opened the pull request:
Fixes #64017 

first add CMD as a Key the same way CMD as been added to KeyModifiersMask
Then replace all KEY::CTRL by KEY::CMD

ajreckof committed:
add CMD_OR_CTRL as a Key and not just a key modifier

mhilbrunner commented:
Bump @groud :)

EricEzaM commented:
cc @bruvzg as they did some recent working on keyboard/input stuff. Not sure if there is a better way of doing this. On the `Input` singleton maybe it would be nice to have `is_modifier_key_pressed`?

bruvzg commented:
`APPLE_STYLE_KEYS` was removed, and there's `CMD_OR_CTRL` mask to specify shortcuts that are `Cmd+` on macOS and `Ctrl+` on other systems.

bruvzg commented:
> Not sure if there is a better way of doing this. On the Input singleton maybe it would be nice to have is_modifier_key_pressed?

What would be the difference from `is_key_pressed`?

PR seems fine, but I would rename `CMD` to `CMD_OR_CTRL` for consistency with current masks, and `#ifdef APPLE_STYLE_KEYS` should be changed to `#ifdef MACOS_ENABLED`.

EricEzaM commented:
> What would be the difference from `is_key_pressed`?

Ah I forgot that we don't distinguish between left/right modifier keys. I also thought it might be helpful as a way to check for if  `CMD_OR_CTRL` is pressed as that is not currently possible. But yeah, just adding `CMD_OR_CTRL` to the key list like you suggest works well.

ajreckof commented:
Just made the changed based on what was discussed here without changing what was done on the bottom ( changed the name of the `CMD` for `CMD_OR_CTRL` and changed the `APPLE_STYLE_KEYS` by `MACOS_ENABLED`. 
It would have been cool to be able to use modifiers but at some point we need to know if the modifier is pressed or no for preview even if there is no input to check modifiers from. This why we need to have it as a key and not just a key modifier

akien-mga commented on core/os/keyboard.cpp:
```diff
@@ -63,12 +63,15 @@ static const _KeyCodeText _keycodes[] = {
 	{Key::CTRL                  ,"Ctrl"},
 #if defined(MACOS_ENABLED)
 	{Key::META                  ,"Command"},
+    {Key::CMD_OR_CTRL           ,"Command"},
```
The indentation is wrong, should use a tab.

I'm surprised the CI checks don't complain about it.

ajreckof replied:
don't know why it put spaces nor why it didn't complain but fixed it

ajreckof committed:
replace the use of CTRL by CMD_OR_CTRL in three places where using CTRL prevented the use of the functionality on mac

on Mac CTRL + Left click = Right click therefore when a functionality needs CTRL + Left click it is not possible on mac. It then forcibly needs to be CMD on mac

akien-mga reviewed (approved):
Looks good! There might be more areas in the engine using `Input::get_singleton()->is_key_pressed(Key::CTRL)` which may benefit from this change.

ajreckof commented:
I'll look into it. Do you think i should create a new pull request or should i had to this one and rename it ?

ajreckof commented:
I have already found at least two (still going through all occurences of calling `is_key_pressed(Key::CTRL)`) other places where using Control instead of command prevents you from using the functionality just as in his case because it needs a left click and using ctrl makes a right click. I think they should also be included in this PR. 

I think after i am done with the `is_key_pressed(Key::CTRL)` i'll look into the `is_ctrl_pressed` as they might also have the same bugs just because people didn't test on mac and didn't think of using `is_command_or_control_pressed()` (or the code was developed before the function was implemented).

akien-mga commented on editor/plugins/canvas_item_editor_plugin.cpp:
```diff
@@ -1897,7 +1897,7 @@ bool CanvasItemEditor::_gui_input_scale(const Ref<InputEvent> &p_event) {
 
 			bool uniform = m->is_shift_pressed();
-			bool is_ctrl = Input::get_singleton()->is_key_pressed(Key::CTRL);
+			bool is_ctrl = m->is_ctrl_pressed();
```
Should it be CTRL or CMD_OR_CTRL here?

ajreckof replied:
ctrl works fine here as you are already dragging when pushing the control button. I just changed it to the event key modifier as it was more in sync with the other declaration just above and there was no reason to call the Input Singleton.

akien-mga replied:
It works, but is it what makes sense semantically? Usually Ctrl being used as modifier on Windows/Linux is intended to be Command on macOS, no?

ajreckof replied:
Most of the time tes but not always more over something have been Cyril for a long time on Godot so people might have already learned it is Godot so I think for now if it is not causing a bug we should not address it right and open an issue to discuss we want to get rid and of all of them / most of them but keep some edge case. 
I think also we might want to discuss wether we would like to add a warning when `is_ctrl_pressed` is used for people not on a mac to understand it is most likely not what they want to use (if there is a way to disable warning for a line)

I think I will be opening an issue that details everything that I found when I'm done looking through all the `is_ctrl_pressed`. There are around 50 of them so it might take me a week or so. As I can see it most use cases can be summarised in a few groups. I think those groups should be moved out from ctrl to command but some edge case might better to keep on ctrl.

akien-mga commented:
Thanks!
